dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -114,11 +121,21 @@ JavaClass loadJavaClass(String className) throws ClassNotFoundException {
     return syntheticRepository.loadClass(className);
   }
 
-  /**
-   * Returns {@link Class} for {@code className} in the input class path using a Java class loader.
-   */
-  Class<?> loadClass(String className) throws ClassNotFoundException {
-    return classLoader.loadClass(className);
+  Class<?> loadClassFromExtensionClassLoader(String className) throws ClassNotFoundException {
+    return extensionClassLoader.loadClass(className);
+  }
+
+  boolean isJavaRuntimeClass(String className) {
+    try {
+      if (className.startsWith("[")) {
+        // Array class
+        return true;
+      }
+      loadClassFromExtensionClassLoader(className);
```

Comment:
Perhaps use findClass or findSystemClass instead. I'm not sure about the exact semantics here, but we may not need to actually load the class at this point.

Reply:
Renamed to `findSystemClass`.

I think using the extension class loader is easy way to distinguish whether a class is from JVM or not.